bl/policy/policy_factory.go

```diff
@@ -7,6 +7,7 @@ import (
 
 	"github.com/datreeio/datree/pkg/cliClient"
 	internal_policy "github.com/datreeio/datree/pkg/policy"
+	"github.com/xeipuuv/gojsonschema"
```

Comment:
We're gonna need to move all of the validations into `/pkg/policy/policy.go` - `validatePoliciesYaml()`

Reply:
Actually, at first I was thinking of adding these validations to `validatePoliciesYaml()` function. But I thought, adding the validation logic to that function would need lots of `yaml.unmarshal` and `for` loops (to check schemas, identifier uniqueness etc.) which we already have in `policy_factory.go` (e.g. [`populateRules()`](https://github.com/datreeio/datree/blob/359680591f6cce4ebc97f0f8e8462b1b2d431c17/bl/policy/policy_factory.go#L69)). So whenever user try to use `--policy-config` , similar operations would run two times (one for validation check and another is for parsing the policies).

We cannot bypass the `populateRules()` function because [`policies`](https://github.com/datreeio/datree/blob/359680591f6cce4ebc97f0f8e8462b1b2d431c17/cmd/test/main.go#L286) given by `evaluationPrerunDataResp.PoliciesJson` need this function. so I thought it would be simpler and would avoid running similar operations two times.

What's your view on it?

Reply:
In my view, in 95% of cases - maintainability > speed (unless there is an actual performance issue)
This may be a bit faster but much harder to maintain, the separation of steps makes it easier to wrap your head around it and make changes to the code later 😊 

step 1 - validate the input
step 2 - use the input (once we know it's clean)